src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/DoNotDisableHttpClientCRLCheck.cs

```diff
@@ -235,8 +235,8 @@ public override void Initialize(AnalysisContext context)
                             }
                             finally
                             {
-                                rootOperationsNeedingAnalysis.Free();
-                                allResults?.Free();
+                                rootOperationsNeedingAnalysis.Free(compilationAnalysisContext.CancellationToken);
+                                allResults?.Free(compilationAnalysisContext.CancellationToken);
```

Comment:
📝 This file is potentially problematic.

* The last time we check the cancellation token is in the call to `PropertySetAnalysis.BatchGetOrComputeHazardousUsages`
* If cancellation occurs after the last check, we will proceed to call `ReportDiagnostic` and finish without throwing an exception, but won't return the objects to the pool

Technically this isn't going to produce corrupt results, but it does mean the pattern doesn't account for cases where we finish successfully.

Reply:
Yeah, I am not too concerned about cancellation firing after analysis but before diagnostic reporting. Bigger concern here is to avoid corruption of data structures.

Reply:
Yes, something that can be handled in a follow-up PR. I don't see any user visible impact for this as IDE and compiler will de-dupe identical diagnostics.